### ISSUE TITLE
[SCREENREADER, KEYBOARD]: Edit address link MUST receive keyboard focus

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#5866]
## Description

When I navigate the "Please confirm your address" screen by keyboard to traverse the content, I can tab to the "Close this modal" button, "Use this address", and "Cancel", but the Edit Address links are not accessible by tab. A screen reader or keyboard user would be unable to edit their address. 

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

- As a keyboard user, I want to be able to tab to all interactive content on the page.

- As a keyboard or screen reader user, I want to be able to access the ability edit my address from the "Please confirm your address" screen.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/profile` in browser
2. Navigate to edit Mailing address
3. Edit mailing address and Update to navigate to the Please confirm your address screen
4. Using only the keyboard, attempt to tab to the Edit address links in order to activate them
5. Verify it is not accessible by keyboard only

### For the paragraph:
**Current:**
```html
<p>
  We couldn’t confirm your address with the U.S. Postal Service. 
  Please verify your address so we can save it to your VA profile. 
  If the address you entered isn’t correct, please <a>edit it</a> 
  or choose a suggested address below.
</p>
```

**Recommend:**
```html
<p>
  We couldn’t confirm your address with the U.S. Postal Service. 
  Please verify your address so we can save it to your VA profile. 
  If the address you entered isn’t correct, please 
  <a href=”link-to-edit”>edit it</a> or choose a suggested address below.
</p>
```
Or, use a `<button>` which is likely the more semantic and appropriate element to use.

### For the “You entered:” part of the modal:

**Current:**
```html
<form>
  <span class="vads-u-font-weight--bold">You entered:</span>
  <div class="vads-u-display--flex vads-u-flex-direction--column vads-u-justify-content--center vads-u-margin-bottom--1p5">
    <label for="userEntered" class="vads-u-margin-top--2 vads-u-display--flex vads-u-align-items--center">
      <div class="vads-u-display--flex vads-u-flex-direction--column">
        <span>123 Main Street</span>
        <span> Portland, ME 04101</span>
        <a>Edit Address</a>
      </div>
    </label>
  </div>
  <button class="usa-button-primary">Use this address</button>
  <button type="button" class="usa-button-secondary">Cancel</button>
</form>
```
- Error: Element div not allowed as child of element label in this context. 
- Error: The value of the for attribute of the label element must be the ID of a non-hidden form control.

**Recommend:**
- Recommend not using`<form>` element here. 
- The label's `for` attribute needs to pair with an `id`. 
- This could be a text display with buttons for edit/confirmation/cancellation. 
- For the `<a href="#">` below, you could optionally use a `<button>`which is likely the more semantic and appropriate element to use.
- The `<dfn>` elements below could all be sr-only. Adding the labels helps to digest the information contextually/semantically.

Here are three different options:
```html
<p aria-describedby=”id-unique”>You entered:</p>
<ul id=”id-unique”>
        <li><dfn>Street address:</dfn> 123 Main Street</li>
        <li><dfn>City:</dfn> Portland, <abbr title=”Maine”>ME</abbr> <dfn>Zip Code:</dfn> 04101</li>
        <li><a href=”link_back”>Edit Address</a></li>
    </ul>
<button>Use this address</button>   <button>Cancel</button>
```
Or:
```html
<dl>
    <dt>You entered:</dt>
    <dd><dfn>Street address:</dfn> 123 Main Street</dd>
    <dd><dfn>City:</dfn> Portland, <abbr title=”Maine”>ME</abbr> <dfn>Zip Code:</dfn> 04101</dd>
    <dd><a href=”#”>Edit Address</a></dd>
</dl>
<button>Use this address</button>   <button>Cancel</button>
```

Or, if you want to use a form:
```html
<form>
  <fieldset>
    <legend>You entered:</legend>
    <ul>
        <li><dfn>Street Address:</dfn> ### Lorem Street</li>
        <li><dfn>City:</dfn> City, <dfn>State:</dfn> ST <dfn>Zip Code:</dfn> 12345</li>
        <li><a href=”#”>Edit Address</a></li>
    </ul>
  </fieldset>
</form>
<button>Use this address</button>   <button>Cancel</button>
```

## WCAG or Vendor Guidance (optional)

* [Focus Order: Understanding SC 2.4.3](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)

* [Ensure that all content can be accessed with the keyboard alone.](https://webaim.org/techniques/keyboard/)

## Screenshots or Trace Logs

![authdex-addressvalidation-a-element](https://user-images.githubusercontent.com/57469/74477303-10e9f980-4e79-11ea-8d13-68bad7877c06.png)

**Comment by Samara-Strauss** [comment]
This sounds relatively straightforward. Triaging over to @bradl3yC. Brad — adding this to sprint goals.

**Comment by bradl3yC** [comment]
@jenstrickland the fix for this was just merged.  It should be available in staging in the next ~30 minutes or so.

**Comment by Samara-Strauss** [comment]
@jenstrickland would super appreciate if you could validate these for us. If it looks good, feel free to close this out!

**Comment by jenstrickland** [comment]
@bradl3yC The `edit it` link in the paragraph is address, but the `Edit this address` link needs a similar treatment in order to be focusable.

**Comment by bradl3yC** [comment]
@Samara-Strauss since this is already merged - we need a follow up ticket to correct @jenstrickland comments I don't see Matt's github id to tag him as well

**Comment by Samara-Strauss** [comment]
@mshea0606 can you please handle ticket creation for this?

**Comment by mshea0606** [comment]
@Samara-Strauss @bradl3yC  Ticket is below:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/6008

**Comment by Samara-Strauss** [comment]
What is the status on this? Can this be closed?

**Comment by jenstrickland** [comment]
As of this morning, the `Edit Address` link is not focusable. Perhaps there are changes to be pushed to staging?

**Comment by bradl3yC** [comment]
@jenstrickland what address did you use to test this?

**Comment by jenstrickland** [comment]
56

**Comment by bradl3yC** [comment]
@jenstrickland apparently there was one more anchor that needed to be swapped to a button.  I just added it to my current screen reader PR since its all 508 related anyway.  As soon as its approved, I will merge to master/staging

**Comment by bradl3yC** [comment]
@jenstrickland @Samara-Strauss the edit links should now be focus-able and the screen reader reading the previous menu should be resolved.  The build should be in staging in the next ~20 minutes or so.

**Comment by Samara-Strauss** [comment]
Validated on staging!

**Comment by jenstrickland** [comment]
![Screen Shot 2020-02-25 at 3 21 35 PM](https://user-images.githubusercontent.com/57469/75285971-99f02180-57e5-11ea-9493-0c6e02e20961.png)

Please adjust the focus outline so that it is visible around the entire area in focus. In the screenshot, it shows I am seeing it cut off on the bottom.

**Comment by bradl3yC** [comment]
@jenstrickland please re-test the edit address focus issue.  Thanks

**Comment by jenstrickland** [comment]
@bradl3yC I checked all the browsers and the focus outline is no longer cut off. I would normally Close and comment, but am not closing this as you asked me not to previously.

**Comment by mshea0606** [comment]
@bradl3yC  Hey, thanks for the update. Is this safe to close now?
